modules/system/config-wfrp4e.js

```diff
@@ -1833,8 +1847,51 @@ WFRP4E.PrepareSystemItems = function() {
             flags: {
                 wfrp4e: {
                     value: 1,
-                    applicationData : {},
+                    applicationData : {
+                        conditionTrigger : "endRound"
+                    },
                     scriptData: [
+                        {
+                            trigger: "manual", 
+                            label: "@effect.name", 
+                            script: `
+                            let actor = this.actor;
+                            let effect = this.effect;
+                            let msg = ""
+                            
+                            let conditionValue = effect.conditionValue;
+                            let msg = "<h2>" + game.i18n.localize("WFRP4E.ConditionName.Stunned") + "</h2>"
+                            let damage = effect.conditionValue;
+                            let scriptArgs = {msg, damage};
+                            await Promise.all(actor.runScripts("preApplyCondition", {effect, data : scriptArgs}))
+                            
+                            let test = await actor.setupSkill(game.i18n.localize("NAME.Endurance"), {appendTitle : " - Oszołomienie"})
+                            await test.roll();
+                            if (test.result.outcome == "success")
+                            {
+                                await actor.removeCondition("stunned", Math.min(test.result.SL, conditionValue));
+                                msg += "<br/>Number of removed Stunned Conditions: " + Math.min(test.result.SL, conditionValue);
+                            }
+                            else
+                            {
+                                msg += "Failed to remove Stunned Condition";
+                            }
+                            let messageData = game.wfrp4e.utility.chatDataSetup(msg);
+                            messageData.speaker = {alias: actor.prototypeToken.name}
+                            await Promise.all(actor.runScripts("applyCondition", {effect, data : {messageData}}))
+                            if (args.suppressMessage)
+                            {
+                                let messageData = game.wfrp4e.utility.chatDataSetup(msg);
+                                messageData.speaker = {alias: this.actor.prototypeToken.name}
+                                messageData.flavor = this.effect.name
+                                return messageData
+                            }
+                            else
+                            {
+                                return this.script.scriptMessage(msg)
+                            }
+                            `
```

Comment:
As said above, I think condition removal should wait, also there is unlocalized strings here.